library/core/class.form.php

```diff
@@ -368,7 +368,7 @@ public function categoryDropDown($FieldName = 'CategoryID', $Options = false) {
                 }
 
                 if ($Category['AllowDiscussions']) {
-                    $Disabled &= $Permission == 'add' && !$Category['PermsDiscussionsAdd'];
+                    $Disabled = $Permission == 'add' && !$Category['PermsDiscussionsAdd'];
```

Comment:
This should probably be |=, but at this point we should just simplify this with another if statement.

```php
if ($Permission == 'add' && !$Category['PermsDiscussionsAdd']) {
    $Disabled = true;
}
```

You want the row that is disabled for some other reason to stay disabled.